### DISCUSSION TITLE
Translators / Translations Needed!

### DISCUSSION THREAD

**Discussion by hargata** [hargata/lubelog#240] [General]
Made this discussion thread to make it easier to coordinate translation efforts, if you're interested in localizing LubeLogger, please read the following documentation:
https://docs.lubelogger.com/Translations

This thread is also where I will post updates regarding when new updates are released that will necessitate further translation.

The translation repository is located at: https://github.com/hargata/lubelog_translations

**Comment by hargata** [comment]
Currently we have a German translation that needs reviewed, I don't speak German, and would greatly appreciate it if someone could review it, thanks!

https://github.com/hargata/lubelog_translations/pull/5

**Reply by rbnhln** [reply]
I've reviewed the german translation and want to make some change suggestions. How do I need to do that?

**Reply by hargata** [reply]
You can leave a comment on his pull request.

**Reply by hargata** [reply]
for now, since we don't have a German translation in the repo yet, once his changes are merged in, in the future, you can make changes to the translation by forking from the repo and then making a pull request.

**Reply by rbnhln** [reply]
Great, found a way to review with the suggestions.

**Comment by 6asc** [comment]
I've just added a French translation, would be great if anyone is able to review it (this is my first translation so there are probably a few mistakes)
https://github.com/hargata/lubelog_translations/pull/12

**Comment by nichwall** [comment]
Hey there! I just discovered LubeLogger and it's pretty nifty! I've helped out a lot with Audiobookshelf, and one of the things we have discussed a lot is managing translations.

Our basic workflow is that users make a copy of the English translation file (which is the source of truth for keys). Any new strings are added to the English source file and a script copies any new keys to the other languages. Contributors can then update translations of any strings they notice are out of date as new strings are added. We require the keys to be in alphabetical order so updates are easier to review.

I recently made an integration workflow that checks that the keys are in order as part of PRs/commits. I also have a workflow that will automatically run the "copy keys script", create a branch, and open a PR for the translations.

Would either of these workflows (or something similar if you can give me more specific instructions) be useful for y'all?

**Comment by ialfarol** [comment]
Hi @hargata, I'm no developer at all, but discovered this piece of software, and would like to implement this in my car repair shop. But, also want to collaborate with the Spanish translation.

Is there a chance to add/change currency support?

**Comment by acmg** [comment]
Hi. 
I've just added a Spanish of Spain translation, feel free to review it: https://github.com/hargata/lubelog_translations/pull/49
Thanks!!

**Comment by Sunsurfer89** [comment]
Moin from Germany,

@hargata while setting up Lubelogger on my NAS (Docker) and start filling data, I noticed that the translations of LubeLogger are not in place everywhere. For example, I discovered the file [wwwroot/js/shared.js](https://github.com/hargata/lubelog/blob/main/wwwroot/js/shared.js) which is providing some popups/modals and all the titles and texts/html code is hardcoded and does not allow a proper translation. 

1. Are there any plans to change the way how the texts are promoted, and is there a way to include placeholders there too, that find their way into the language JSON files?
I guess that would make the app more robust and user-friendly.

While speaking about that: 
2. Are there any ambitions to use translation services such as transifex or weblate (or similar things) to make the translation contribution in a more controlled manner? 

Looking forward to your comments and plans. I would love to help where I can because Lubelogger shall become my master car recording tool. (I want to get rid of another paid Andorid app in favor of LubeLogger 😉)

**Reply by Sunsurfer89** [reply]
... and in addition: Is there a specific reason that the default language file [wwwroot/defaults/en_US.json](https://github.com/hargata/lubelog/blob/main/wwwroot/defaults/en_US.json) is a single line code file? Wouldn't it be better to have one line per key:value pair? Just for better read-ability and for easier usage as master file for other languages?

**Reply by hargata** [reply]
You're supposed to use the translation editor instead of editing it line by line: https://docs.lubelogger.com/Misc/Translations#creating-your-own-translation

Regarding the hardcoded texts in the popups, we plan to move away from the JS-based popups sometime next year because the JS popups are really hard to shimmy the translation layer

**Reply by Sunsurfer89** [reply]
Many thanks for your answer. 

I'm aware of the Editor but the problem/burden, that I discovered, is the following:

Example:
The file [main/Views/Vehicle/_ReportHeader.cshtml](https://github.com/hargata/lubelog/blob/main/Views/Vehicle/_ReportHeader.cshtml) contains the string "Distance Traveled". Of course, this string pattern exists in the default language file [wwwroot/defaults/en_US.json](https://github.com/hargata/lubelog/blob/main/wwwroot/defaults/en_US.json) with key-value-pair: `"Distance_Traveled":"Distance Traveled"`
However, if we look for example into the [lubelog_translations/blob/main/Europe/de.json](https://github.com/hargata/lubelog_translations/blob/main/Europe/de.json) translation, this key-value pair is not defined, so Lubelogger will do a fall-back to the default translation. Which is fine.

But what I'm trying to emphasize is the fact, that the language files are relatively quickly out of sync. To update/change translations, from my little perspective, there are a few steps necessary... which could be at least better controlled and finally maintained by the mentioned translation tools. I expect that this would be very beneficial for the tool reputation to have such a centralized translation setup.
And finally, no code is necessary for an own translation editor... 😉 Just my little opinion. 😇

Hope, I was able to show what I mean and where I see some room for improvement. But maybe there is a good reason to do it differently?!

**Comment by cr4zyduck** [comment]
Just Added the Catalan Translation: https://github.com/hargata/lubelog_translations/pull/93

**Comment by Merinorus** [comment]
Here is an update for the French translation: https://github.com/hargata/lubelog_translations/pull/94

**Comment by igoraraujocosta** [comment]
Hi! I've submitted a complete update to the pt_BR (Brazilian Portuguese) 
translation: https://github.com/hargata/lubelog_translations/pull/100

- Translated all 241 missing keys (was at 415/655)
- Fixed 2 mistranslated keys and removed 1 obsolete key
- Standardized terminology across old and new entries to match existing 
  community translations (e.g. "odometer"→"hodômetro")

Separately, I wanted to ask about two things:

1. Regarding the hardcoded JS popups mentioned earlier in this thread 
   (shared.js) — is there already a tracking issue for the move away from 
   them, or should contributors just watch this discussion for updates?

2. Would the maintainers be open to using a platform like Crowdin for 
   translation management instead of the current fork+PR workflow? 
   Crowdin has API/MCP support, which could enable automated 
   contribution and review workflows, and would also solve the 
   key-ordering/sync issues raised earlier in this thread by nichwall.

Happy to help test or contribute to either effort if useful.

**Comment by Ikuo-310** [comment]
Hi,

I created a Japanese localization for LubeLogger:

https://github.com/Ikuo-310/lubelogger-ja-jp

It is localized for Japanese car and motorcycle users, so some wording differs from the original English UI.

If you are interested in adding Japanese to the official translations, please let me know how I should proceed.

Thank you.
